support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -135,9 +131,12 @@ public HttpResponse doSubmit(StaplerRequest request) throws IOException, Servlet
      * REST endpoint to submit the input.
      */
     @RequirePOST
-    public HttpResponse doProceed(StaplerRequest request) throws IOException, ServletException, InterruptedException {
+    public HttpResponse doProceed(StaplerRequest request, User user) throws IOException, ServletException, InterruptedException {
```

Comment:
`doProceed` is an implicit `@WebMethod` so I am not sure what the impact is of adding a new parameter (even if it is not marked `@QueryParameter`).

Anyway this seems unnecessary. The check for `User.current()` could just be in `doProceed` to begin with, right? Just as it is in `doAbort`.